plugins/generator-1.15.2/forge-1.15.2/templates/plant.java.ftl

```diff
@@ -394,6 +394,60 @@ import net.minecraft.block.material.Material;
             </#if>
         </#if>
 
+		<#if (data.canBePlacedOn?size > 0) || hasCondition(data.placingCondition)>
+			<#if (data.canBePlacedOn?size > 0) && data.plantType != "growapable">
+			@Override public boolean isValidGround(BlockState state, IBlockReader worldIn, BlockPos pos) {
+				Block block = state.getBlock();
+				return
+				<#list data.canBePlacedOn as canBePlacedOn>
+					block == ${mappedBlockToBlockStateCode(canBePlacedOn)}.getBlock()
+					<#if canBePlacedOn?has_next>||</#if>
+				</#list>;
+			}
+			</#if>
+
+			@Override public boolean isValidPosition(BlockState state, IWorldReader worldIn, BlockPos pos) {
+				BlockPos blockpos = pos.down();
+				BlockState blockstate = worldIn.getBlockState(blockpos);
+				Block block = blockstate.getBlock();
+				<#if hasCondition(data.placingCondition)>
+				boolean additionalCondition = true;

```

Comment:
You can declare the additionalCondition variable directly at line 419

Reply:
Done.